### PR TITLE
parquet-cli: Patch snappy version on parquet-cli pom.xml

### LINKED ISSUE #93135: parquet-cli: patch "095c78fec3378189296d38fede1255b0a4d05fd4" does not work
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
kylemart@PowerBookPro ~ % brew config
HOMEBREW_VERSION: 3.3.10-51-g26941f0
ORIGIN: https://github.com/Homebrew/brew
HEAD: 26941f0bfd40bec21af461c1b0ed55267bf430cf
Last commit: 5 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 2c560a62559502bfeae1c7e59e2512eaff96c453
Core tap last commit: 18 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.32.0 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.1-arm64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2.1
Rosetta 2: false


kylemart@PowerBookPro ~ % brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /Users/kylemart/.pyenv/shims/python3.10-config
  /Users/kylemart/.pyenv/shims/python-config
  /Users/kylemart/.pyenv/shims/python3-config
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

I'm using an M1 MacBook Pro. When trying to run `parquet` (parquet-cli installed via Brew) commands, I'm encountering exceptions related to snappy. A recent commit made to this repo should have patched the issue, but it seems to have not worked for me. Here is the PR: https://github.com/Homebrew/homebrew-core/pull/92069.

### What happened (include all command output)?

kylemart@PowerBookPro ~ % parquet head /Users/kylemart/Downloads/userdata1.parquet
Exception in thread "main" org.xerial.snappy.SnappyError: [FAILED_TO_LOAD_NATIVE_LIBRARY] no native library is found for os.name=Mac and os.arch=aarch64
	at org.xerial.snappy.SnappyLoader.findNativeLibrary(SnappyLoader.java:361)
	at org.xerial.snappy.SnappyLoader.loadNativeLibrary(SnappyLoader.java:195)
	at org.xerial.snappy.SnappyLoader.loadSnappyApi(SnappyLoader.java:167)
	at org.xerial.snappy.Snappy.init(Snappy.java:69)
	at org.xerial.snappy.Snappy.<clinit>(Snappy.java:46)
	at org.apache.parquet.hadoop.codec.SnappyDecompressor.decompress(SnappyDecompressor.java:62)
	at org.apache.parquet.hadoop.codec.NonBlockedDecompressorStream.read(NonBlockedDecompressorStream.java:51)
	at java.base/java.io.DataInputStream.readFully(DataInputStream.java:201)
	at java.base/java.io.DataInputStream.readFully(DataInputStream.java:172)
	at org.apache.parquet.bytes.BytesInput$StreamBytesInput.toByteArray(BytesInput.java:286)
	at org.apache.parquet.bytes.BytesInput.toByteBuffer(BytesInput.java:237)
	at org.apache.parquet.bytes.BytesInput.toInputStream(BytesInput.java:246)
	at org.apache.parquet.column.impl.ColumnReaderBase.readPageV1(ColumnReaderBase.java:680)
	at org.apache.parquet.column.impl.ColumnReaderBase.access$300(ColumnReaderBase.java:57)
	at org.apache.parquet.column.impl.ColumnReaderBase$3.visit(ColumnReaderBase.java:623)
	at org.apache.parquet.column.impl.ColumnReaderBase$3.visit(ColumnReaderBase.java:620)
	at org.apache.parquet.column.page.DataPageV1.accept(DataPageV1.java:120)
	at org.apache.parquet.column.impl.ColumnReaderBase.readPage(ColumnReaderBase.java:620)
	at org.apache.parquet.column.impl.ColumnReaderBase.checkRead(ColumnReaderBase.java:594)
	at org.apache.parquet.column.impl.ColumnReaderBase.consume(ColumnReaderBase.java:735)
	at org.apache.parquet.column.impl.ColumnReaderImpl.consume(ColumnReaderImpl.java:30)
	at org.apache.parquet.column.impl.ColumnReaderImpl.<init>(ColumnReaderImpl.java:47)
	at org.apache.parquet.column.impl.ColumnReadStoreImpl.getColumnReader(ColumnReadStoreImpl.java:82)
	at org.apache.parquet.io.RecordReaderImplementation.<init>(RecordReaderImplementation.java:271)
	at org.apache.parquet.io.MessageColumnIO$1.visit(MessageColumnIO.java:147)
	at org.apache.parquet.io.MessageColumnIO$1.visit(MessageColumnIO.java:109)
	at org.apache.parquet.filter2.compat.FilterCompat$NoOpFilter.accept(FilterCompat.java:177)
	at org.apache.parquet.io.MessageColumnIO.getRecordReader(MessageColumnIO.java:109)
	at org.apache.parquet.hadoop.InternalParquetRecordReader.checkRead(InternalParquetRecordReader.java:136)
	at org.apache.parquet.hadoop.InternalParquetRecordReader.nextKeyValue(InternalParquetRecordReader.java:225)
	at org.apache.parquet.hadoop.ParquetReader.read(ParquetReader.java:132)
	at org.apache.parquet.hadoop.ParquetReader.read(ParquetReader.java:136)
	at org.apache.parquet.cli.BaseCommand$1$1.advance(BaseCommand.java:363)
	at org.apache.parquet.cli.BaseCommand$1$1.<init>(BaseCommand.java:344)
	at org.apache.parquet.cli.BaseCommand$1.iterator(BaseCommand.java:342)
	at org.apache.parquet.cli.commands.CatCommand.run(CatCommand.java:73)
	at org.apache.parquet.cli.Main.run(Main.java:155)
	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:76)
	at org.apache.parquet.cli.Main.main(Main.java:185)

### What did you expect to happen?

I expected the command to not throw an exception.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
Note: I performed the following steps on an M1 Mac.

Here are the steps I performed:
1. Updated Homebrew to the latest version.
2. Ran `brew install parquet-cli`
3. Downloaded this sample parquet file to my Downloads folder: https://github.com/Teradata/kylo/blob/master/samples/sample-data/parquet/userdata1.parquet.
4. Ran `parquet head /Users/kylemart/Downloads/userdata1.parquet`
```

### PR DESCRIPTION
**Pull request by shanielh**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Fixes #93135

### CODE REVIEW

File: Formula/parquet-cli.rb
Comment (carlocab) [inline Formula/parquet-cli.rb:61]:
```suggestion
end

__END__
```
This is in the style guide (and makes it the start of the patch easier to distinguish visually).

**Review by carlocab — COMMENTED** [review]
Thanks; just one small suggestion. Can you bump the `revision` when you make this change?

Please squash these changes into your existing commit rather than add new commits.

File: Formula/parquet-cli.rb
Comment (carlocab) [inline Formula/parquet-cli.rb:36]: Also, has this issue been reported upstream?

Comment (shanielh) [reply Formula/parquet-cli.rb:36]: This is not a problem upstream, the parquet-cli project will use equivalent parquet-hadoop package (version)

**Comment by shanielh** [comment]
> Thanks; just one small suggestion. Can you bump the `revision` when you make this change?

How do I bump the revision?

**Comment by carlocab** [comment]
Add a line that says `revision 1` right after `license`.

File: Formula/parquet-cli.rb
Comment (carlocab) [inline Formula/parquet-cli.rb:62]:
```suggestion
```
Only a single newline please. Thanks!

Comment (shanielh) [reply Formula/parquet-cli.rb:62]: Done

**Comment by shanielh** [comment]
> Add a line that says `revision 1` right after `license`.

Done

**Comment by carlocab** [comment]
Thanks again.

The test on Linux is failing, which means this can't be merged as-is, unfortunately. The failure may well be unrelated to your changes, but merging this with a failing `brew test` will delete the Linux bottle, which we can't do.

**Comment by iMichka** [comment]
The Linux error was already here before, and I wanted to fix it because it can often been seen in other pull requests when recursive dependencies get tested. I found this upstream error: https://issues.apache.org/jira/browse/PARQUET-2104 which looks similar

**Comment by carlocab** [comment]
Hm, that's not good. I guess our options here are:
1. Skip the test on Linux (this probably means shipping broken bottles)
2. Make this formulae macOS-only until the upstream issue is fixed (will this break existing Linux installs, or are they already broken without further changes?)
3. Wait until we have a fix on Linux to merge this change (this also means shipping broken bottles on ARM)

I don't think any of these options are good, but presumably there is a least-bad alternative.

**Comment by shanielh** [comment]
Hi @carlocab , what are we going to do?

**Comment by iMichka** [comment]
@shanielh I would go for solution 2. Install statistics show that there were only 16 installations on Linux in the last year. The project does not look really active either, with the latest commits being from 2019, and not that many stars on GitHub.

Please add `depends_on :macos` and and a link to this PR as a comment please.

For reference the issue we see on Linux is the following:

```
2022-02-02T08:40:12.1071164Z ##[group]Full test parquet-cli output
2022-02-02T08:40:12.1071901Z [32m==>[0m [1mTesting parquet-cli[0m
2022-02-02T08:40:12.1111524Z [31mError:[0m test failed
2022-02-02T08:40:12.1112258Z cp -p /github/home/.cache/Homebrew/downloads/6955f2d3cbcafb29a1424c124c1326dc0263f67471536320aa610b4ba5fa37d1--homebrew.parquet /tmp/parquet-cli--test-parquet-20220202-21084-1h42tz1/homebrew.parquet
2022-02-02T08:40:12.1113361Z [34m==>[0m [1m/home/linuxbrew/.linuxbrew/Cellar/parquet-cli/1.12.0_1/bin/parquet cat /tmp/parquet-cli-test-20220202-21084-1ydxq9n/homebrew.parquet[0m
2022-02-02T08:40:12.1142777Z Picked up _JAVA_OPTIONS: -Duser.home=/github/home/.cache/Homebrew/java_cache -Djava.io.tmpdir=/tmp
2022-02-02T08:40:12.1144901Z Exception in thread "main" java.lang.NoSuchMethodError: 'shaded.parquet.org.apache.avro.Schema org.apache.parquet.avro.AvroSchemaConverter.convert(org.apache.parquet.schema.MessageType)'
2022-02-02T08:40:12.1145613Z 	at org.apache.parquet.cli.util.Schemas.fromParquet(Schemas.java:89)
2022-02-02T08:40:12.1146069Z 	at org.apache.parquet.cli.BaseCommand.getAvroSchema(BaseCommand.java:405)
2022-02-02T08:40:12.1146508Z 	at org.apache.parquet.cli.commands.CatCommand.run(CatCommand.java:66)
2022-02-02T08:40:12.1146892Z 	at org.apache.parquet.cli.Main.run(Main.java:155)
2022-02-02T08:40:12.1147246Z 	at org.apache.hadoop.util.ToolRunner.run(ToolRunner.java:76)
2022-02-02T08:40:12.1147586Z 	at org.apache.parquet.cli.Main.main(Main.java:185)
```

**Comment by shanielh** [comment]
@iMichka ✅

LMK if you need anything else

**Review by iMichka — APPROVED** [review]

**Comment by iMichka** [comment]
Thanks @shanielh !

**Pull request closed without merging**
